plugins/modules/postgresql_set.py

```diff
@@ -194,12 +194,54 @@
 # To allow to set value like 1mb instead of 1MB, etc:
 LOWERCASE_SIZE_UNITS = ("mb", "gb", "tb")
 
+# GUC_LIST_QUOTE parameters list for each version where they changed (from PG_REQ_VER).
+# It is a tuple of tuples as we need to iterate it in order.
+PARAMETERS_GUC_LIST_QUOTE = (
+    (140000, (
+        'local_preload_libraries',
+        'search_path',
+        'session_preload_libraries',
+        'shared_preload_libraries',
+        'temp_tablespaces',
+        'unix_socket_directories'
+    )),
+    (90400, (
+        'local_preload_libraries',
+        'search_path',
+        'session_preload_libraries',
+        'shared_preload_libraries',
+        'temp_tablespaces'
+    )),
+)
+
+
 # ===========================================
 # PostgreSQL module specific support methods.
 #
 
 
-def param_get(cursor, module, name):
+def param_is_guc_list_quote(server_version, name):
+    for guc_list_quote_ver, guc_list_quote_params in PARAMETERS_GUC_LIST_QUOTE:
+        if server_version >= guc_list_quote_ver:
+            return name in guc_list_quote_params
+    return False
+
+
+def param_guc_list_unquote(value):
+    # unquote GUC_LIST_QUOTE parameter (each element can be quoted or not)
+    value_unquoted = []
+    for v in value.split(','):
+        v = v.strip()  # strip whitespaces at start/end
+        if v and v[0] == v[-1] == '"':
+            # is quoted -> strip quotes
+            value_unquoted.append(v[1:-1])
+        else:
+            # is not quoted -> no changes
+            value_unquoted.append(v)
+    return ', '.join(value_unquoted)
```

Comment:
```suggestion
    ', '.join([v.strip('" ') for v in value.split(',')])
```

Reply:
how about this simplification?

Reply:
It is correct, because with PARAMETERS_GUC_LIST_QUOTE list we are sure the parameter is quoted and if PostgreSQL is not bugged there are always two `"`.
Thank you very much.